packages/eas-cli/src/analytics.ts

```diff
@@ -63,30 +87,62 @@ export async function setUserDataAsync(
 }
 
 export async function flushAsync(): Promise<void> {
-  if (client) {
-    await client.flush();
+  if (rudderstackClient) {
+    rudderstackClient.flush();
+  }
+  if (amplitudeClient) {
+    await amplitudeClient.flush();
   }
 }
 
 export function logEvent(name: string, properties: Record<string, any> = {}) {
-  if (client) {
-    ensureUserIdentified();
+  if (!amplitudeClient && !rudderstackClient) {
+    return;
+  }
+  ensureUserIdentified();
+
+  if (amplitudeClient) {
     const { userId, deviceId } = identifyData ?? {};
-    client.logEvent({
+    amplitudeClient.logEvent({
       event_type: name,
       event_properties: properties,
       ...(userId && { user_id: userId }),
       ...(deviceId && { device_id: deviceId }),
       ...getContext(),
     });
   }
+
+  if (rudderstackClient) {
+    const { userId, deviceId } = identifyData ?? {};
+
+    const identity = userId
+      ? { userId, anonymousId: undefined }
+      : {
+          userId: undefined,
+          anonymousId: deviceId ?? uuidv4(),
+        };
+
+    rudderstackClient.track({
+      event: name,
+      properties,
+      ...identity,
+      ...(deviceId && { device_id: deviceId }),
+      ...getContext(),
+    });
+  }
 }
 
 function ensureUserIdentified() {
-  if (client && !userIdentifyCalled && identifyData) {
-    client.identify(identifyData.userId, identifyData.deviceId, identifyData.identify);
-    userIdentifyCalled = true;
+  if (!(rudderstackClient && amplitudeClient && !userIdentifyCalled && identifyData)) {
```

Comment:
returning here if both clients are not initialized. this should not, but could break the amplitude identify flow. any pref for the implementation here? i think the other option is to rewrite it to continue if any client is initialized. i think i prefer how it is here because we do want both clients to be initialized.